numpy/core/arrayprint.py

```diff
@@ -402,39 +399,38 @@ def _object_format(o):
     return fmt.format(o)
 
 def repr_format(x):
-    if isinstance(x, (np.str_, np.bytes_)):
+    if isinstance(x, (np.str_, np.bytes_)) and _format_options['legacy'] > 125:
         return repr(x.item())
     return repr(x)
 
 def str_format(x):
-    if isinstance(x, (np.str_, np.bytes_)):
+    if isinstance(x, (np.str_, np.bytes_)) and _format_options['legacy'] > 125:
         return str(x.item())
```

Comment:
Not tha tit matters, but I think this is unnecessary (and thus easier to remove)?!  (`item()` works for the old version, too).

Reply:
Sure, if the output is identical, there is no reason to have it!